src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -1060,20 +1060,27 @@ function findValidPasteFileTarget(targetFolder: ExplorerItem, fileToPaste: { res
 
 export function incrementFileName(name: string, isFolder: boolean): string {
 	const separators = '[\\.\\-_]';
+	const maxNumber = 9007199254740991;
 
 	// file.1.txt=>file.2.txt
 	let suffixFileRegex = RegExp('(.*' + separators + ')(\\d+)(\\..*)$');
 	if (!isFolder && name.match(suffixFileRegex)) {
 		return name.replace(suffixFileRegex, (match, g1?, g2?, g3?) => {
-			return g1 + strings.pad(parseInt(g2) + 1, g2.length) + g3;
```

Comment:
Can we have a helper method for this?
Since this code seems like it is duplicated 4 times?

Reply:
It's bit different for every case since it formats the string in different formats, not sure if there is a sense in trying to extract that to helper function.